[l10n] Add Chinese (Taiwan) (zh-TW) locale

nusr opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

<!-- You can use `## Changelog` to create a description for this change in the next release. -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

LukasTy commented:
@nusr I'm curious, maybe you will be able to clarify: is it common to use different dialects of the written Chinese language in different countries speaking it?
We are currently discussing the addition of the Arabic language, and given that most of the written Arabic uses the standard form, there is no need to have many very similar localizations. 🤷

nusr commented:
> @nusr I'm curious, maybe you will be able to clarify: is it common to use different dialects of the written Chinese language in different countries speaking it? We are currently discussing the addition of the Arabic language, and given that most of the written Arabic uses the standard form, there is no need to have many very similar localizations. 🤷

Chinese writing is mainly divided into Traditional Chinese and Simplified Chinese.
Traditional Chinese characters are closer to the original form of Chinese characters. They often can carry and convey the cultural information, historical allusions, and philosophical ideas contained in Chinese characters more completely. However, they are relatively complex to write and are mainly used in Hong Kong, China, and Taiwan, China.
Simplified Chinese is a simplified version of Traditional Chinese and is mainly used on the Chinese mainland.
The main difference between the two lies in the writing form, while the language and grammar are basically the same.
The relationship between Traditional Chinese and Simplified Chinese is similar to that between Old English and Modern English.

LukasTy commented:
Thank you for the clarification. 🙏 
It seems that in the case of Chinese, there is merit to having multiple localizations. 👌
